packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -379,23 +374,12 @@ export function getCurrentEventPriority(): * {
   return getEventPriority(currentEvent.type);
 }
 
-export function logRecoverableError(
-  config: ErrorLoggingConfig,
-  error: mixed,
-): void {
-  const onRecoverableError = config;
-  if (onRecoverableError !== null) {
-    // Schedule a callback to invoke the user-provided logging function.
-    scheduleCallback(IdlePriority, () => {
-      onRecoverableError(error);
-    });
-  } else {
-    // Default behavior is to rethrow the error in a separate task. This will
-    // trigger a browser error event.
-    scheduleCallback(IdlePriority, () => {
-      throw error;
-    });
-  }
+export function logRecoverableError(error: mixed): void {
+  // Default behavior is to rethrow the error in a separate task. This will
+  // trigger a browser error event.
+  queueMicrotask(() => {
+    throw error;
```

Comment:
Any worries about the Safari bug that can execute this on unclean stack if iframe is added?

Reply:
Sigh. Yeah.

Reply:
I don't really know what the workaround is, though, other than to avoid queueMicrotask. I'm tempted to leave it as-is and leave it to Safari to fix their shit.

Reply:
Maybe a user space one. It doesn't catch on break etc. but gives you something in the DEV console and in prod.

```
console.error(...);
window.dispatchEvent(new ErrorEvent(...));`
```

Reply:
Is one of those options better than the other?

Reply:
It seems to me that queueMicrotask doesn't actually solve the problem, does it? Because the rethrown error will be logged before the microtask. So in the default case, the order is still hard error followed by recovered error. So there's no difference from postTask.

Where as the user space log + dispatch would be recovered error followed by hard error.

Reply:
If you have a recovered error and then a useLayoutEffect or anything causing a second commit in the same task, that then errors because the recovering didn't quite work. Then recovered errors should ideally come first.

Reply:
Ah when we were discussing error ordering I was thinking it was other errors that happened in subsequent events, not the hard error that we throw at the top when there's no error boundary. (I don't think about those usually because I assume most apps have a top-level error boundary.) That makes sense.